tools/AutoTuner/src/autotuner/distributed.py

```diff
@@ -863,23 +912,44 @@ def parse_arguments():
         " training stderr\n\t2: also print training stdout.",
     )
 
-    arguments = parser.parse_args()
-    if arguments.mode == "tune":
-        arguments.algorithm = arguments.algorithm.lower()
+    args = parser.parse_args()
+    if args.mode == "tune":
+        args.algorithm = args.algorithm.lower()
         # Validation of arguments
-        if arguments.eval == "ppa-improv" and arguments.reference is None:
+        if args.eval == "ppa-improv" and args.reference is None:
             print(
                 '[ERROR TUN-0006] The argument "--eval ppa-improv"'
                 ' requires that "--reference <FILE>" is also given.'
             )
             sys.exit(7)
 
-    arguments.experiment += f"-{arguments.mode}-{DATE}"
+    args.experiment += f"-{args.mode}-{DATE}"
 
-    if arguments.timeout is not None:
-        arguments.timeout = round(arguments.timeout * 3600)
+    # Convert time to seconds
+    if args.timeout_per_trial is not None:
+        args.timeout_per_trial = round(args.timeout_per_trial * 3600)
+    if args.timeout is not None:
+        args.timeout = round(args.timeout * 3600)
+
+    # Calculate timeout based on cpu_budget
+    if args.cpu_budget != -1:
+        args.timeout = round(args.cpu_budget / os.cpu_count() * 3600)
+        args.timeout_per_trial = round(
+            args.cpu_budget / (args.jobs * args.resources_per_trial) * 3600
+        )
+        overall_timeout = min(args.timeout, args.timeout_per_trial)
+        if args.mode == "tune":
+            template = calculate_expected_numbers(overall_timeout, args.samples)
+        else:
+            template = calculate_expected_numbers(overall_timeout, 1)
+        print(template)
+        if not args.yes:
+            ans = input("Are you sure you want to proceed? (y/n): ")
```

Comment:
Add a print before this line or include in the input statement that the `-y/--yes` is available if the user does not wish to be asked again.